crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -256,6 +259,11 @@ def provided_with_enabled(feature,
             unless repos[:inactive][r.arch.to_sym].include?(r.name)
               repos[:inactive][r.arch.to_sym].push(r.name)
             end
+
+            next if r.required == "mandatory"
+            unless repos[:optional][r.arch.to_sym].include?(r.name)
+              repos[:optional][r.arch.to_sym].push(r.name)
+            end
```

Comment:
This block is after `next if r.active?`, which is strange. It looks like we only put inactive optional repos in that variable.

Reply:
Yes, that's true. If they are active we do not need any more info, and if they are inactive then there's a problem unless we know they are optional.
(It just seemed nicer to me to test for "optional" then "not mandatory", but in the end it does not matter)